src/widget/input/completion.rs

```diff
@@ -22,18 +22,36 @@ impl Completion {
     }
 
     /// Process input and update the completion state
-    pub fn process(&mut self, input: &str, users: &[User]) {
-        if input.starts_with('/') {
+    pub fn process(&mut self, input: &str, users: &[User], channels: &[String]) {
+        let last_char_is_space = input
+            .chars()
+            .last()
+            .map(|last| last.is_whitespace())
+            .unwrap_or_default();
+        let words: Vec<&str> = input.split_whitespace().collect();
+        let is_channel = !last_char_is_space
+            && words
+                .last()
+                .map(|word| word.starts_with('#'))
+                .unwrap_or_default();
```

Comment:
Let's push this complexity into `Text::process` which can delegate to `process_users` or `process_channels` internally.